tests/unit_test/data_processor/data_processor2_test.py

```
@@ -1289,3 +1290,219 @@ def test_delete_collection_data_no_records():
+
+        assert "errors" in result
+        assert len(result["errors"]) == 1
+        assert result["errors"][0]["index"] == 1
+        assert "Invalid data" in result["errors"][0]["error"]
+        mock_objects.insert.assert_called_once()
+        assert len(mock_objects.insert.call_args[0][0]) == 1  # One valid doc inserted
+
+
+def test_bulk_insert_fails(valid_payload):
+    with patch("kairon.shared.cognition.data_objects.CollectionData.objects") as mock_objects, \
+         patch("kairon.shared.data.collection_processor.DataProcessor.validate_collection_payload") as mock_validate, \
```

Ruff (0.12.2)

Local variable `mock_validate` is assigned to but never used

Remove assignment to unused variable `mock_validate`

(F841)